### PR TITLE
Adds events to Colossus and process them

### PR DESCRIPTION
**Pull request by vwraposo**
The SC Broadcaster was previously calling the broadcaster directly, and it was processing the sku changes immediately. 
This caused a lot of problems when a lot of changes happened in the catalog (for example multiple stores re-indexing the catalog).

Therefore, this PR makes broadcaster send the SKU changed as an event to Colossus, and then process it and send the particular events. This will allow the processing to be slower and more safe for the whole platform.

### CODE REVIEW

File: node/typings.ts [hunk 1]
```diff
@@ -1,10 +1,18 @@
-import { ServiceContext, RecorderState, IOClients, ParamsContext } from '@vtex/api'
+import { RecorderState, IOClients, EventContext, ServiceContext, ParamsContext } from '@vtex/api'
 
 declare global {
   type Context = ServiceContext<IOClients, State, ParamsContext>
+  interface BroadcasterEventContext extends EventContext<IOClients, State> {
+    key: string
```
Comment (tlgimenes) [inline node/typings.ts]: this is already present in eventContext right ?

File: node/middlewares/pushNotification.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { notification } from './../utils/event'
+
+export async function pushNotification(ctx: Context, next: () => Promise<any>) {
+  const { clients: { events } } = ctx
+
+  events.sendEvent('', notification, ctx.state.payload)
```
Comment (tlgimenes) [inline node/middlewares/pushNotification.ts]: we should await this before returning 200, don't we ?

File: node/utils/event.ts [hunk 3]
```diff
@@ -4,3 +4,4 @@ export const skuChanged = eventName('sku')
 export const productChanged = eventName('product')
 export const brandChanged = eventName('brand')
 export const categoryChanged = eventName('category')
+export const notification = eventName('notification')
```
Comment (tlgimenes) [inline node/utils/event.ts]: why not using a more descriptive name, like `broadcaster notification`, or `rawNotification`

Comment (vwraposo) [reply node/utils/event.ts]: The name is: `broadcaster.notification`

**Pull request closed without merging**